system/jlib/jstring.cpp

```diff
@@ -937,27 +937,58 @@ StringBuffer & StringBuffer::replace(char oldChar, char newChar)
 // Copy source to result, replacing all occurrences of "oldStr" with "newStr"
 StringBuffer &replaceString(StringBuffer & result, size_t lenSource, const char *source, size_t lenOldStr, const char* oldStr, size_t lenNewStr, const char* newStr)
 {
-    if (lenSource)
+    // Only copy if the source and search (oldStr) strings are non-empty
+    if (lenSource && lenOldStr)
```

Comment:
I think these need to be handled separately. This works for lenSource == 0, but if lenSource > 0 and lenOldStr == 0 then you need to copy the source to the result before returning it.

Reply:
Fixed. When lenSource > 0 and lenOldStr == 0 the source is copied to result.